src/EditorFeatures/CSharp/Formatting/CSharpFormattingInteractionService.cs

```diff
@@ -107,7 +107,8 @@ public async Task<ImmutableArray<TextChange>> GetFormattingChangesOnPasteAsync(D
             var fallbackOptions = _globalOptions.GetCSharpSyntaxFormattingOptions();
             var options = await _indentationManager.GetInferredFormattingOptionsAsync(document, fallbackOptions, explicitFormat: true, cancellationToken).ConfigureAwait(false);
             var service = document.GetRequiredLanguageService<ISyntaxFormattingService>();
-            return await service.GetFormattingChangesOnPasteAsync(document, textSpan, options, cancellationToken).ConfigureAwait(false);
+            var documentSyntax = await ParsedDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
```

Comment:
`CreateAsync` might be nice as an extension to `Document`

Reply:
Might be, or even a regular instance method.

Reply:
Actually, I prefer the factory pattern currently. We can switch later.